packages/frontend/src/pages/campaigns/filters/checkboxes-filter/index.tsx

```diff
@@ -0,0 +1,101 @@
+import { Template } from "@carrot-kpi/sdk";
+import { Checkbox, Typography } from "@carrot-kpi/ui";
+import React, { ChangeEvent, useCallback, useEffect, useState } from "react";
+import { FiltersLoading } from "../loading";
+import {
+    CheckboxesObj,
+    generateGroupId,
+    getDefaultCheckboxes,
+    getDefaultSelected,
+} from "../utils";
+
+interface CheckboxesFilterProps {
+    items: Template[];
+    groupId: string;
+    title: string;
+    loading: boolean;
+    selected: Set<number>;
+    setSelected: (newSelected: Set<number>) => void;
+    defaultChecked: (item: Template) => boolean;
+}
+
+export const CheckboxesFilter = ({
+    items,
+    title,
+    loading,
+    selected,
+    setSelected,
+    defaultChecked,
+    groupId,
+}: CheckboxesFilterProps) => {
+    const [checkboxes, setCheckboxes] = useState<CheckboxesObj>({});
+
+    const initializeCheckboxes = useCallback(() => {
+        if (items.length === 0) return;
+        if (Object.keys(checkboxes).length !== 0) return;
+
+        const defaultCheckboxes = getDefaultCheckboxes(items, defaultChecked);
+        setCheckboxes(defaultCheckboxes);
+        setSelected(getDefaultSelected(defaultCheckboxes));
+    }, [checkboxes, defaultChecked, items, setSelected]);
+
+    const getHtmlDataId = (element: HTMLInputElement) => {
```

Comment:
This function not being in a `useCallback` is a bit of an issue on rerenders. See comment below.